### ISSUE TITLE
Did you experiment with different batch sizes in training?

### ISSUE THREAD

**Issue by stefan-kohnen** [simonalexanderson/StyleGestures#29]
Hey there,

Again, thank you very much for sharing your code ❤️ 

A batch size of 80 is the default for your model. @ghenter mentioned in issue #9 that you have run the GPU fairly to the maximum capacity in terms of memory during experiments in training of MoGlow/StyleGestures. Does this mean that you do not have experimented with batch sizes > 80 in early stages? And if you actually did experiment with bigger batch sizes, did you see some effect on the synthesized gestures in terms of humanlikeness and appropriateness?

Same question for batch sizes < 80: did you experiment with smaller batch sizes in early training stages? Any effects on quality of synthesized gestures?

A bit of background to this question: I forked your repository and for training I extended the input feature space with BERT feature vectors of the corresponding transcripts. The cluster on which I run the model training has a GPU memory capacity that forces me to use a batch size of approx. 10 because the BERT features actually need by some magnitudes more memory than the speech features. 
The generated samples which result from training with this batch size look like being "fast-forwarded", so they move way faster than for example the generated samples which are generated with the original feature space (i.e., your code). Also the motion and poses are not really humanlike. 

So in all my question is intended so that I get to know if there is something wrong with my code and concept or if these motion artifacts arise from the fact that I use a small batch size.

Looking forward to your reply. It would help me a great deal! =)

Best regards,
Stefan

**Comment by simonalexanderson** [comment]
Hi @stefan-kohnen ,
Adding BERT features sound very interesting and I would love to see what the results will show. I guess you have seen our work with Gesticulator (Kucherenko et al 2020), which has a different architecture. We have not experimented so much with batch sizes, but tend to use the maximum for our gpus, which typically is 50-100. 10 seems low to me. You may experiment with accumulating gradients over several batches (search for "accumulating gradients") or checkpointing the rnn in the coupling layer: https://github.com/prigoyal/pytorch_memonger/blob/master/tutorial/Checkpointing_for_PyTorch_models.ipynb. The later should give a large memory improvement with only a few lines of code. And please share your findings :-).

**Comment by stefan-kohnen** [comment]
Hey @simonalexanderson,
Thank you for your answer. Yes, I saw Gesticulator :+1: Accumulating Gradients and Gradient Checkpointing both seem like a good idea, thanks for the hint. 
I trained your model with batch size of 10 to see if this gives the same effect as with the input feature space which I extended by BERT features (the one I called "fast-forwarding").
The clear answer: **No**, it does not. The motion of the generated gestures look as smooth as with a batch size of 80. Of course, they most likely differ in other aspects in comparison to the training with a batch size of 80, I didn't check on these. But they definitely do not show the same behavior as with the integration of BERT features (i.e., "fast-fowarding").
